packages/dds/tree/src/feature-libraries/sequence-field/rebase.ts

```diff
@@ -146,6 +165,78 @@ function rebaseMark<TNodeChange>(
             return clone(currMark);
         }
         default:
-            fail("Not implemented");
+            fail(`Unsupported mark type: ${baseType}`);
+    }
+}
+
+function handleCurrAttach<T>(
+    currMark: Attach<T>,
+    factory: MarkListFactory<T>,
+    lineageRequests: LineageRequest<T>[],
+    offset: number,
+) {
+    const rebasedMark = clone(currMark);
+    factory.pushContent(rebasedMark);
+    lineageRequests.push({ mark: rebasedMark, offset });
```

Comment:
Why do we need to defer the construction of the lineage information here. IOW, why not have this function take the base `RevisionTag` and just do `mark.lineage.push({ revision, offset: offset });` (modulo the undefined `mark.lineage` array case)?

Reply:
Originally I thought we might need to capture information about later base marks. That didn't end up being necessary, but I think we may need to do that in the future. Currently the only reason for doing it this way is that when processing this attach, we do not yet know if the base change has a detach immediately following this attach's target position. If the offset is zero and there is no detach following this attach, we do not want to add a lineage event.

Reply:
If I'm following correctly, your last sentence implies that we would want to add a lineage event if the offset were zero but there is a detach (in the base change) following this attach. Am I understanding correctly that this is because we want to be able to differentiate an insert at the left edge of a deleted region that used to target the gap before the deleted node from an insert that just targets the "crater" gap left by that same deletion?

Reply:
Yes

Reply:
A small comment explaining that would be nice.